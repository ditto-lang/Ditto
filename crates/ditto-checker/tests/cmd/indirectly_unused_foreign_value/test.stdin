module Test exports (five)

five : Float = 5.0

unused = unused_impl

foreign unused_impl: (Int) -> Bool
