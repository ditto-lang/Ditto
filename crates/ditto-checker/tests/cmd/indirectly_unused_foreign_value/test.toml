bin.name = "ditto-checker-testbin-check-module"
args = ["indirectly_unused_foreign_value.ditto"]
fs.sandbox = true
