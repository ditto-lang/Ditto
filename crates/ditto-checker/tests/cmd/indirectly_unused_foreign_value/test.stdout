{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {},
    "constructors": {},
    "values": {
      "five": {
        "doc_comments": [],
        "doc_position": 0,
        "value_type": {
          "type": "PrimConstructor",
          "data": "Float"
        }
      }
    }
  },
  "types": {},
  "constructors": {},
  "values": {
    "unused": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 48,
        "end_offset": 54
      },
      "expression": {
        "expression": "ForeignVariable",
        "data": {
          "span": {
            "start_offset": 57,
            "end_offset": 68
          },
          "variable_type": {
            "type": "Function",
            "data": {
              "parameters": [
                {
                  "type": "PrimConstructor",
                  "data": "Int"
                }
              ],
              "return_type": {
                "type": "PrimConstructor",
                "data": "Bool"
              }
            }
          },
          "variable": "unused_impl"
        }
      }
    },
    "five": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 28,
        "end_offset": 32
      },
      "expression": {
        "expression": "Float",
        "data": {
          "span": {
            "start_offset": 43,
            "end_offset": 46
          },
          "value": "5.0",
          "value_type": {
            "type": "PrimConstructor",
            "data": "Float"
          }
        }
      }
    }
  },
  "values_toposort": [
    "unused",
    "five"
  ]
}